### PR TITLE
Update profiler example code

### PR DESCRIPTION
**Pull request by kenfehling**
## Description ##
Documentation

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: docs/faq/perf.md [hunk 1]
```diff
@@ -228,12 +228,12 @@ See [example/profiler](https://github.com/dmlc/mxnet/tree/master/example/profile
 for complete examples of how to use the profiler in code, but briefly, the Python code looks like:
 
 ```
-    mx.profiler.profiler_set_config(mode='all', filename='profile_output.json')
-    mx.profiler.profiler_set_state('run')
+    mx.profiler.set_config(profile_all=True, filename='profile_output.json')
+    mx.profiler.set_state('run')
```
Comment (szha) [inline docs/faq/perf.md]: This API change shouldn't have happened. @cjolivier01 would you mind adding `profiler_set_state` back to fix the breaking change in #8972? cc'd @eric-haibin-lin who discovered the issue in #8972, @piiswrong who merged the breaking change.

Comment (chinakook) [reply docs/faq/perf.md]: ```https://github.com/apache/incubator-mxnet/tree/master/example/profiler``` also need to change.

**Comment by piiswrong** [comment]
@cjolivier01 
I thought we added backward compatible wrappers?

File: docs/faq/perf.md (hunk 1 above)
Comment (cjolivier01) [reply docs/faq/perf.md]: Changing the AP*I to the new code is desired

**Comment by cjolivier01** [comment]
yes, that one was missed. fallback is merged in already

**Pull request merged**